Decred Wallet/Features/History/TransactionHistoryViewController.swift

```diff
@@ -55,140 +59,139 @@ class TransactionHistoryViewController: UIViewController {
     
     override func viewDidAppear(_ animated: Bool) {
         if !AppDelegate.walletLoader.isSynced {
-            print(" wallet not synced on history")
+            print("wallet not synced on history")
             return
         }
-        self.visible = true
-        if (self.FromMenu){
-            refreshControl.showLoader(in: self.tableView)
-            loadTransactions()
-            FromMenu = true
-        }
+        self.refreshControl.showLoader(in: self.tableView)
+        loadTransactions()
     }
     
-    override func viewDidDisappear(_ animated: Bool) {
-        super.viewDidDisappear(animated)
-        self.visible = false
-        dismiss(animated: true, completion: nil)
+    func initFilterBtn() {
+        self.btnFilter.initMenu(self.filterMenu) { [weak self] index, value in
+            self?.applyTxFilter(currentFilter: self!.filters[index])
+        }
     }
     
-    @objc func handleRefresh(_ refreshControl: UIRefreshControl) {
-        self.loadTransactions()
+    func applyTxFilter(currentFilter: Int) {
+        refreshControl.showLoader(in: self.tableView)
+        filteredItems.removeAll()
+        
+        switch currentFilter{
```

Comment:
We could reduce a few lines here. Once we have a Filter enum then we can have a computed var that returns its title. e.g
enum Filter: Int {
 case all = 0
 var title: NSLocalizedString
}
Also with filter enum then our method will expect a filter i.e `applyTxFilter(_ filter: Filter)`. Therefore we could just use a few lines for this e.g
self.filterActive = filter != .none
self.btnFilter.setTitle(filter.locaizedString.appending(filteredItems.count))
self.filteredItems = self.transactions.filter{$0.Direction == filter.rawValue && $0.Type == filter.globalConstant}
This way testing becomes easy and also easy for whoever would want to inject some new logic in there